pkg/api/alerts.go

```diff
@@ -0,0 +1,40 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package api
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/NYTimes/gziphandler"
+)
+
+func Alerts(conf *Config, p *provider, updateMetrics func(handler, code string, duration float64)) http.Handler {
+	hf := corsWrapper(conf, alertsHandler(p, updateMetrics))
+	return gziphandler.GzipHandler(hf)
+}
+
+func alertsHandler(p *provider, updateMetrics func(handler, code string, duration float64)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		statusCode := "400"
+		begin := time.Now()
+		defer func() {
+			updateMetrics("/api/v1/rules", statusCode, time.Since(begin).Seconds())
```

Comment:
Yeah, this can be part of some future PR. I think when this was being implemented, we wanted to keep things straight for the first version. Now it makes sense to make this smarter.